rmgpy/tools/observablesregression.py

```diff
@@ -112,32 +112,72 @@ def __init__(self, title='', old_dir='', new_dir='', observables=None, expt_data
+        # define chemkin file path for old and new models
+        old_chemkin_path = os.path.join(old_dir, 'chem_annotated.inp')
+        new_chemkin_path = os.path.join(new_dir, 'chem_annotated.inp')
+        old_species_dict_path = os.path.join(old_dir, 'species_dictionary.txt')
+        new_species_dict_path = os.path.join(new_dir, 'species_dictionary.txt')
+
+        surface = False
+        if os.path.exists(os.path.join(old_dir, 'chem_annotated-surface.inp')) and \
+            os.path.exists(os.path.join(old_dir, 'chem_annotated-gas.inp')):
+            surface = True
+            old_chemkin_path = os.path.join(old_dir, 'chem_annotated-gas.inp')
+            new_chemkin_path = os.path.join(new_dir, 'chem_annotated-gas.inp')
+            old_surface_chemkin_path = os.path.join(old_dir, 'chem_annotated-surface.inp')
+            new_surface_chemkin_path = os.path.join(new_dir, 'chem_annotated-surface.inp')
+            ck2cti = True  # can't create a surface model from memory yet, must load from yaml
+
+        old_species_list, old_reaction_list = load_chemkin_file(
+            old_chemkin_path,
+            old_species_dict_path,
+            old_transport_path
+        )
+        new_species_list, new_reaction_list = load_chemkin_file(
+            new_chemkin_path,
+            new_species_dict_path,
+            new_transport_path
+        )
+
+        old_surface_species_list = None
+        new_surface_species_list = None
+        new_surface_reaction_list = None
+        old_surface_reaction_list = None
+        if surface:
+            old_surface_species_list, old_surface_reaction_list = load_chemkin_file(
+                old_surface_chemkin_path,
+                old_species_dict_path,
+                old_transport_path
+            )
+            new_surface_species_list, new_surface_reaction_list = load_chemkin_file(
+                new_surface_chemkin_path,
+                new_species_dict_path,
+                new_transport_path
+            )
 
         self.old_sim = Cantera(species_list=old_species_list,
-                               reaction_list=old_reaction_list,
-                               output_directory=old_dir)
+                            reaction_list=old_reaction_list,
+                            output_directory=old_dir,
+                            surface_species_list=old_surface_species_list,
+                            )
         self.new_sim = Cantera(species_list=new_species_list,
-                               reaction_list=new_reaction_list,
-                               output_directory=new_dir)
+                            reaction_list=new_reaction_list,
+                            output_directory=new_dir,
+                            surface_species_list=new_surface_species_list,
+                            )
 
         # load each chemkin file into the cantera model
-        if not ck2cti:
+        if not ck2cti:  # TODO make this work for surfaces
```

Comment:
Could you open/link the issue on the GitHub page to track this?

Reply:
Here's the new issue: https://github.com/ReactionMechanismGenerator/RMG-Py/issues/2600